tutorials/scripting/c_sharp/c_sharp_variant.rst

```diff
@@ -0,0 +1,156 @@
+.. _doc_c_sharp_variant:
+
+C# Variant
+==========
+
+For a detailed explanation of Variant in general, see the :ref:`doc_variant_class` section in the core and modules tutorial.
+
+``Godot.Variant`` is used to represent the Godot's native :ref:`Variant <doc_variant_class>` type. Any Variant-compatible type can be converted from/to it.
+We recommend avoiding ``Godot.Variant`` unless it is necessary to interact with untyped engine APIs.
+Take advantage of C#'s type safety when possible.
+
+Any of ``Variant.As{TYPE}`` methods or the generic ``Variant.As<T>`` method can be used to convert
+a ``Godot.Variant`` to a C# type. Since the ``Godot.Variant`` type contains implicit conversions
+defined for all the supported types calling these methods directly is usually not necessary.
+
+Use ``CreateFrom`` method overloads or the generic ``Variant.From<T>`` method to convert a C# type
+to a ``Godot.Variant``.
+
+.. note::
+
+    Since the Variant type in C# is a struct, it can't be null. To create a "null"
+    Variant use the ``default`` keyword or the parameterless constructor.
+
+Variant-compatible types
+------------------------
+
+* All the `built-in value types <https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/built-in-types-table>`_,
+  except ``decimal``, ``nint`` and ``nuint``.
+* ``string``.
+* Classes derived from :ref:`GodotObject <class_Object>`.
+* Collections types defined in the ``Godot.Collections`` namespace.
+
+Full list of Variant types and their equivalent C# type:
```

Comment:
Is there a way to view this list in the class reference? Maintaining such a list manually is prone to being outdated, in my experience.

Reply:
Considering how resistant we are to adding new types to `Variant` I don't think it's likely to get outdated.

I understand your concern though, but I'd rather not separate the list from the rest of the C# Variant documentation. I think it's convenient to have a single page that explains how the `Variant` type works in C#, together with the list of "Variant-compatible types" so that whenever we use that phrase users can tell what we mean.